src/gadgets/hash.rs

```diff
@@ -141,12 +145,8 @@ impl<F: PrimeFieldBits> HashChip<F> {
             |mut region| {
                 self.config.selector.enable(&mut region, 0)?;
 
-                let input_cell = region.assign_advice(
-                    || "input",
-                    self.config.input,
-                    0,
-                    || input.value().cloned(),
-                )?;
+                let input_cell =
+                    input.copy_advice(|| "input", &mut region, self.config.input, 0)?;
```

Comment:
Changed that from a call to `region.assign_advice()`. That also adds a copy constraint, which is essential for soundness!